src/main/java/htsjdk/variant/vcf/VCFContigHeaderLine.java

```diff
@@ -76,9 +76,14 @@ public Integer getContigIndex() {
         return contigIndex;
     }
 
+    /**
+     * Get the SAMSequenceRecord that corresponds to this VCF header line.
+     * @return The SAMSequenceRecord containing the ID, length, assembly, and index of this contig. Returns null if the
+     * contig header line does not have a length.
+     */
 	public SAMSequenceRecord getSAMSequenceRecord() {
 		final String lengthString = this.getGenericFieldValue("length");
-		if (lengthString == null) throw new TribbleException("Contig " + this.getID() + " does not have a length field.");
+		if (lengthString == null) return null;
```

Comment:
SAMSequenceRecord has an `UNKNOWN_SEQUENCE_LENGTH` value specifically for the case where the length is missing. It would be preferable to use that here and return a `SAMSequenceRecord`, rather than returning null in cases where it previously would have thrown.

Reply:
It might be worth adding a comment noting the SAMSequenceRecord.isSameSequence treats a record created this way (with a length of `UNKNOWN_SEQUENCE_LENGTH`) as matching another sequence record with the same name but a valid length.